src/core/Tokenizer.js

```diff
@@ -29,8 +30,12 @@ export default class Tokenizer {
         this.OPEN_PAREN_REGEX = this.createParenRegex(cfg.openParens);
         this.CLOSE_PAREN_REGEX = this.createParenRegex(cfg.closeParens);
 
-        this.PLAIN_VARIABLE_REGEX = this.createVariableRegex(cfg.variableTypes, "[a-zA-Z0-9._$]+");
-        this.QUOTED_VARIABLE_REGEX = this.createVariableRegex(cfg.variableTypes, this.createStringPattern(cfg.stringTypes));
+        this.INDEXED_PLACEHOLDER_REGEX = this.createPlaceholderRegex(cfg.indexedPlaceholderTypes);
+        this.PLAIN_NAMED_PLACEHOLDER_REGEX = this.createPlaceholderRegex(cfg.namedPlaceholderTypes, "[a-zA-Z0-9._$]+");
+        this.STRING_NAMED_PLACEHOLDER_REGEX = this.createPlaceholderRegex(
```

Comment:
I don't really like the name `PLAIN_NAMED`. Perhaps better `IDENT_NAMED` to emphasize that the name is an identifier.

Speaking of identifiers though, the regex `[a-zA-Z0-9._$]+` isn't fully covered by tests - we don't test for the special characters `.`, `_` and `$`. I wonder, do we even really need to support `.` and `$`.